source/configuration_manual/fts/flatcurve.rst

```diff
@@ -0,0 +1,225 @@
+.. _fts_backend_flatcurve:
+
+Flatcurve FTS Engine
+====================
+
+.. versionadded:: 2.4.0;3.0.0
+
+Requires Xapian 1.4+ `<https://xapian.org/download/>`_ to work properly.
+
+This is a Dovecot FTS plugin to enable message indexing using the
+[Xapian](https://xapian.org/) Open Source Search Engine Library.
+
+The plugin relies on Dovecot to do the necessary stemming. It is intended
+to act as a simple interface to the Xapian storage/search query
+functionality.
+
+This driver supports match scoring and substring matches (on by default),
+which means it is RFC 3501 (IMAP4rev1) compliant. This driver does not
+support fuzzy searches.
+
+The driver passes all of the ImapTest search tests.
+`<https://imapwiki.org/ImapTest>`_
+
+Enabling flatcurve is designed to be as easy as adding these lines:
+
+.. code-block:: none
+
+    mail_plugins = $mail_plugins fts fts_flatcurve
+    plugin {
+        fts = flatcurve
+    }
+
+Optional parameters
+-------------------
+
+.. note:: The default settings should be fine in most scenarios.
+
+.. dovecot_plugin:setting:: fts_flatcurve_commit_limit
+  :plugin: fts-flatcurve
+  :default: 500
+  :values: @uint, !set to 0 to use the Xapian default
+
+  Commit database changes after this many documents are updated. Higher commit
+  limits will result in faster indexing for large transactions (i.e. indexing
+  a large mailbox) at the expense of high memory usage. The default value
+  should be sufficient to allow indexing in a 256 MB maximum size process.
+
+.. dovecot_plugin:setting:: fts_flatcurve_max_term_size
+  :plugin: fts-flatcurve
+  :default: 30
+  :values: @uint, !max: 200
+
+  The maximum number of characters in a term to index.
+
+.. dovecot_plugin:setting:: fts_flatcurve_min_term_size
+  :plugin: fts-flatcurve
+  :default: 2
+  :values: @uint
+
+  The minimum number of characters in a term to index.
+
+.. dovecot_plugin:setting:: fts_flatcurve_optimize_limit
+  :plugin: fts-flatcurve
+  :default: 10
+  :values: @uint, !set to 0 to disable
+
+  Once the database reaches this number of shards, automatically optimize the
+  DB at shutdown.
+
+.. dovecot_plugin:setting:: fts_flatcurve_rotate_size
+  :plugin: fts-flatcurve
+  :default: 5000
+  :values: @uint, !set to 0 to disable rotation
```

Comment:
As mentioned in the code MR, this should be size.

Reply:
it actually is a count, renamed from fts_flatcurve_rotate_size to fts_flatcurve_rotate_count